cpp/src/strings/regex/regexec.cu

```diff
@@ -104,14 +109,14 @@ std::unique_ptr<reprog_device, std::function<void(reprog_device*)>> reprog_devic
   auto d_end = d_ptr + (classes_count * sizeof(reclass_device));
   // place each class and append the variable length data
   for (int32_t idx = 0; idx < classes_count; ++idx) {
-    reclass& h_class = h_prog.class_at(idx);
+    auto h_class = h_prog.class_at(idx);
```

Comment:
To prevent the copying of `literals` (as the previous code did), should this not be `auto const& h_class = ...;`? Or is that affordable?

Reply:
You are correct. Good catch.